sklearn/dummy.py

```diff
@@ -581,10 +581,9 @@ def fit(self, X, y, sample_weight=None):
             if sample_weight is None:
                 self.constant_ = np.median(y, axis=0)
             else:
-                self.constant_ = [
-                    _weighted_percentile(y[:, k], sample_weight, percentile_rank=50.0)
-                    for k in range(self.n_outputs_)
-                ]
```

Comment:
When looping across `y[:,k]`, are you not effectively asking for `axis=1` ? I thought we calculated `_weighted_percentile` across `axis=0` ?